packages/component/src/hooks/useEmojiFromStyles.js

```diff
@@ -0,0 +1,51 @@
+/*eslint no-useless-escape: "off"*/
+/*eslint require-unicode-regexp: "off" */
+import useStyleOptions from './useStyleOptions';
+
+function escapeRegexp(emoticon) {
+  return emoticon.replace(/[\\^$*+?.()|[\]{}]/g, '\\$&');
+}
+
+export default function useEmojiFromStyles() {
+  const [{ emojiAutocorrect: autocorrect, emojiList: customEmojiList }] = useStyleOptions();
+
+  const emojiAutocorrect = autocorrect || false;
+
+  const emojiUnicodeMap = customEmojiList || {
```

Comment:
Please put the default list in the style options.

Reply:
do you mean `defaultStyleOptions`? Just making sure.

Reply:
Retreating on this one. I think you might not able to put in `defaultStyleOptions.js`.

I think you might want to put it in `Composer.js`, i.e. patch the list as early as possible.

You can search `patchedXxx` in `Composer.js`.